test/src/main/java/org/axonframework/test/aggregate/ResultValidatorImpl.java

```diff
@@ -95,7 +96,7 @@ public ResultValidatorImpl(List<EventMessage<?>> publishedEvents,
 
     @Override
     public ResultValidator<T> expectEvents(EventMessage... expectedEvents) {
-        this.expectEvents((Object[]) expectedEvents);
+        this.expectEvents((Object[]) (Stream.of(expectedEvents).map(Message::getPayload).toArray()));
```

Comment:
Can't we do another map call in between, which casts the payload to object?
So, something like this:
```
this.expectEvents(Stream.of(expectedEvents).map(Message::getPayload).map(payload -> (Object) payload).toArray());
```

Reply:
Actually, I don't think we need a cast at all, this should be sufficient:

```
this.expectEvents(Stream.of(expectedEvents).map(Message::getPayload).toArray());
```

Reply:
The cast has indeed become superfluous. Pushed a new commit that removes it.